app/src/main/java/ca/llamabagel/transpo/data/TripsRepository.kt

```diff
@@ -25,23 +25,29 @@ import javax.inject.Singleton
 class TripsRepository @Inject constructor(
     private val database: TransitDatabase,
     private val apiService: TripsService,
-    private val sharedPreferences: SharedPreferences
+    private val sharedPreferences: SharedPreferences,
+    private val dispatcherProvider: CoroutinesDispatcherProvider
 ) {
     private val cachedResults: MutableMap<StopCode, ConflatedBroadcastChannel<ApiResponse>> = mutableMapOf()
     private val cachedStopCodes: MutableMap<StopId, StopCode> = mutableMapOf()
 
-    suspend fun getResultCache(stopId: StopId): ConflatedBroadcastChannel<ApiResponse> = withContext(Dispatchers.IO) {
-        val stopCode =
-            cachedStopCodes.getOrPut(stopId, { database.stopQueries.getStopById(stopId).executeAsOne().code })
+    suspend fun getResultCache(stopId: StopId): ConflatedBroadcastChannel<ApiResponse> {
+        val stopCode = cachedStopCodes.getOrPut(
+            stopId, {
+                withContext(dispatcherProvider.io) { database.stopQueries.getStopById(stopId).executeAsOne().code }
+            }
+        )
 
-        cachedResults.getOrPut(stopCode, { ConflatedBroadcastChannel() })
+        return cachedResults.getOrPut(stopCode, { ConflatedBroadcastChannel() })
     }
 
     fun clearCacheFor(stopId: StopId) {
-        cachedStopCodes[stopId]?.let(cachedResults::remove)
+        if (cachedStopCodes[stopId] != null) {
```

Comment:
Are you checking if the key exists or if the value is not null?
```suggestion
        if (cachedStopCodes.containsKey(stopId))
```